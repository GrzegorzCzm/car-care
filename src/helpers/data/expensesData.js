const expenses = [
  {
    id: 0, date: '2018-04-19', item: 'tires', cost: 35.45, description: 'some details here', address: 'dummy address', position: { lat: 51.109372, lng: 16.958476 },
  },
  {
    id: 1, date: '2018-08-13', item: 'some long name here', cost: 85, description: 'some more details here', address: 'dummy second address', position: { lat: 51.109372, lng: 16.958476 },
  },
  {
    id: 2, date: '2019-07-10', item: 'wheels', cost: 1450.65, description: 'some more details here', address: 'dummy third address', position: { lat: 51.109372, lng: 16.958476 },
  },
];


export default expenses;
