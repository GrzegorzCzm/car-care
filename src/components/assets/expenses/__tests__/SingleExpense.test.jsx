import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import expensesData from '../../../../helpers/data/expensesData';

import SingleExpenseWithStyles from '../SingleExpense';


describe('SingleExpense', () => {
  const props = {
    expense: expensesData[0],
  };

  it('should renders correctly', () => {
    const wrapper = shallow(<SingleExpenseWithStyles {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should show Suspense with Spinner before map component is loaded', () => {
    const wrapper = mount(<SingleExpenseWithStyles {...props} />);

    expect(wrapper.find('Suspense').length).toBe(1);
    expect(wrapper.find('Spinner').length).toBe(1);
  });
});
