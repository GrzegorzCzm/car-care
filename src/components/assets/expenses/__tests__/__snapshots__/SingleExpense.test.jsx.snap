// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleExpense should renders correctly 1`] = `
<SingleExpense
  classes={
    Object {
      "card": "SingleExpense-card-1",
      "details": "SingleExpense-details-4",
      "detailsText": "SingleExpense-detailsText-5",
      "heading": "SingleExpense-heading-2",
      "secondaryHeading": "SingleExpense-secondaryHeading-3",
    }
  }
  expense={
    Object {
      "address": "dummy address",
      "cost": 35.45,
      "date": "2018-04-19",
      "description": "some details here",
      "id": 0,
      "item": "tires",
      "position": Object {
        "lat": 51.109372,
        "lng": 16.958476,
      },
    }
  }
/>
`;
